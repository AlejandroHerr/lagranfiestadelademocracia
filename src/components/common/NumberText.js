import React from 'react';
import PropTypes from 'prop-types';

import { formatDecimal } from '../../utils/format';

import NormalText from './NormalText';

const NumberText = ({ children, decimals, ...props }) => (
  <NormalText {...props}>{formatDecimal(children, decimals).toString()}</NormalText>
);

NumberText.propTypes = {
  children: PropTypes.number.isRequired,
  decimals: PropTypes.number,
};

NumberText.defaultProps = {
  decimals: 0,
};

export default NumberText;
