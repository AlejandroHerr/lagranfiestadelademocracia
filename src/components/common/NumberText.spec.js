import React from 'react';
import renderer from 'react-test-renderer';

import NumberText from './NumberText';

describe('components/component/NumberText', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NumberText>{1 + 1 / 3}</NumberText>).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with decimales', () => {
    const tree = renderer.create(<NumberText decimals={7}>{1 + 1 / 3}</NumberText>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
