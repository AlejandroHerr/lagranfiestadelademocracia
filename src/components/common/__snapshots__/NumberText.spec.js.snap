// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/component/NumberText renders correctly 1`] = `
<div
  className="sc-bdVaJa sc-htpNat jBGZHO"
  fontFamily="body"
>
  1
</div>
`;

exports[`components/component/NumberText renders correctly with decimales 1`] = `
<div
  className="sc-bdVaJa sc-htpNat jBGZHO"
  fontFamily="body"
>
  1.3333333
</div>
`;
